### PR TITLE
bf(S3C-4057): Handle timestamp conflicts during ingestion

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: libV2/config/index.js [hunk 1]
```diff
@@ -271,7 +271,7 @@ class Config {
             warp10Conf.hosts = [{
                 host: _loadFromEnv('WARP10_HOST', config.warp10.host),
                 port: _loadFromEnv('WARP10_PORT', config.warp10.port, _typeCasts.int),
-                nodeId: parsedConfig.nodeId,
+                nodeId: _loadFromEnv('WARP10_NODE_ID', config.warp10.nodeId),
```
Comment (vrancurel) [inline libV2/config/index.js:274]: will it be the node host ?

File: libV2/utils/timestamp.js [hunk 2]
```diff
@@ -19,16 +19,16 @@ class InterpolatedClock {
...
 
-    getTs() {
-        const ts = Date.now();
+    getTs(timestamp) {
+        const ts = timestamp !== undefined ? timestamp : Date.now();
         if (ts === this._now) {
             // If this is the same millisecond as the last call
             this._step += 1;
-            return ts * 1000 + (this._step - 1);
+            return convertTimestamp(ts) + (this._step - 1);
```
Comment (vrancurel) [inline libV2/utils/timestamp.js:27]: what if the next one also collide ?

File: tests/functional/v2/task/testIngestShards.js [hunk 3]
```diff
@@ -144,5 +144,46 @@ describe('Test IngestShards', function () {
         );
         series[0].values.forEach(val => assert.strictEqual(val.id, undefined));
     });
+
+    it('should increment microseconds for duplicate timestamps', async () => {
```
Comment (vrancurel) [inline tests/functional/v2/task/testIngestShards.js:148]: Do we test the case where there are >2 events conflicting on the same ts ?

File: libV2/utils/timestamp.js (hunk 2 above)
Comment (tmacro) [reply libV2/utils/timestamp.js:27]: `this._step` is incremented each time.

File: libV2/config/index.js (hunk 1 above)
Comment (tmacro) [reply libV2/config/index.js:274]: `config.warp10.nodeId`  will be the host warp 10 is on (used for `node` label). The top level `nodeId` is the host the utapi process is running on (used for `origin` label)

File: tests/functional/v2/task/testIngestShards.js (hunk 3 above)
Comment (tmacro) [reply tests/functional/v2/task/testIngestShards.js:148]: I'll add one

**Review by vrancurel — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**